pkg/controller/utils/clientwrapper.go

```diff
@@ -74,7 +74,7 @@ func NewClientWithMetricsOrDie(mgr manager.Manager, ctrlrName hivev1.ControllerN
 		log.WithError(err).Fatal("unable to initialize metrics wrapped client")
 	}
 
-	return dc
+	return client.WithFieldOwner(dc, "hive1-"+string(ctrlrName))
```

Comment:
I didn't mention this earlier but I'm glad you were able to find these paths and number them for debugging. I suppose some day we should try and streamline the ways we touch k8s objects - we might even succeed at lessening the vendored deps.

Reply:
Totally agree it would be great to consolidate to a much smaller set of k8s-client-y code.

Off the top, it seems like the major categories we have are:
- Controllers talking to hub
- Controllers talking to spoke
- CLIs (imageset, provision & deprovision pods; other standalone `hiveutil`z) talking to hub

I can see why these should go through different paths high in the call stack, but once it comes to actually creating the kube client from a config, I'm not sure there's any reason for them to be divergent.

That said, we would want to be _very_ thorough in testing the critical paths -- like clustersync! -- if we swap those guts out. It's hard to justify replacing foundational pieces of components that are at this point so well hardened and proven.

Reply:
Hardened, proven, and entirely opaque... I agree and think a better future is possible